src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -405,6 +407,16 @@ int sd_dhcp_client_set_vendor_class_identifier(
         return free_and_strdup(&client->vendor_class_identifier, vci);
 }
 
+
+int sd_dhcp_client_set_user_class(
+                sd_dhcp_client *client,
+                const char *user_class) {
```

Comment:
humm, I really don't like this. Please always pass normalized data around, and parse everything at configuration parsing time, not at the time we want to use it. This means, this really should take a `char **` (i.e. a strv) instead of a string that still needs to be parsed.

Sorry, but this is a no-go.